util/repin_all.sh

```diff
@@ -1,10 +1,21 @@
-#!/bin/bash -eu
+#!/usr/bin/env bash
+set -eu
 
-# Re-generates all files which may need to be re-generated after changing crate_universe.
+# Normalize working directory to root of repository.
+cd "$(dirname "${BASH_SOURCE[0]}")"/..
+
+# Ensure Nix is present and `nix-command` and `flakes` features are enabled.
+if ! command -v nix &> /dev/null; then
+    echo "'nix' is missing from 'PATH'"
+    echo "Visit https://nixos.org/download for installation instructions."
+    exit 1
+fi
+export NIX_CONFIG="$NIX_CONFIG\nexperimental-features = nix-command flakes"
```

Comment:
This fails if `$NIX_CONFIG` isn't set...

Just using `${NIX_CONFIG:-}` doesn't work because nix doesn't appear to like a plain leading `\n`.

I changed it like this:
```diff
diff --git a/util/repin_all.sh b/util/repin_all.sh
index c7dba917..8fc9a0a2 100755
--- a/util/repin_all.sh
+++ b/util/repin_all.sh
@@ -10,7 +10,12 @@ if ! command -v nix &> /dev/null; then
     echo "Visit https://nixos.org/download for installation instructions."
     exit 1
 fi
-export NIX_CONFIG="$NIX_CONFIG\nexperimental-features = nix-command flakes"
+new_nix_config="experimental-features = nix-command flakes"
+if [[ -z "${NIX_CONFIG:-}" ]]; then
+  export NIX_CONFIG="${new_nix_config}"
+else
+  export NIX_CONFIG="${NIX_CONFIG}\n${new_nix_config}"
+fi
 
 # Re-generates all files which may need to be re-generated after changing crate_universe.
 bazel run //crate_universe/3rdparty:crates_vendor
```

but it still doesn't seem to work with me on my ARM Mac:

```console
% util/repin_all.sh   
WARNING: Running Bazel server needs to be killed, because the startup options are different.
Starting local Bazel server and connecting to it...
INFO: Invocation ID: bd09af97-4b47-46c0-86d0-fba066034fcc
INFO: Analyzed target //crate_universe/3rdparty:crates_vendor (114 packages loaded, 351 targets configured).
INFO: Found 1 target...
Target //crate_universe/3rdparty:crates_vendor up-to-date:
  bazel-bin/crate_universe/3rdparty/crates_vendor.sh
INFO: Elapsed time: 3.890s, Critical Path: 0.01s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Running command line: bazel-bin/crate_universe/3rdparty/crates_vendor.sh
Loading: 0 packages loaded
Loading: 0 packages loaded
 checking cached actions
error: flake 'path:/Volumes/Source/github.com/bazelbuild/rules_rust/examples/nix_cross_compiling/nix' does not provide attribute 'packages.aarch64-darwin.bazel.rust', 'legacyPackages.aarch64-darwin.bazel.rust' or 'bazel.rust'
INFO: Repository nix_rust instantiated at:
  /Volumes/Source/github.com/bazelbuild/rules_rust/examples/nix_cross_compiling/WORKSPACE.bazel:17:17: in <toplevel>
  /Volumes/Source/github.com/bazelbuild/rules_rust/examples/nix_cross_compiling/bazel/nix_repositories.bzl:213:26: in nix_repositories
  /private/var/tmp/_bazel_dwagnerhall/ce461197ed3d17b80eddc01283ea4d6f/external/rules_nixpkgs_core/nixpkgs.bzl:876:27: in nixpkgs_flake_package
Repository rule _nixpkgs_flake_package defined at:
  /private/var/tmp/_bazel_dwagnerhall/ce461197ed3d17b80eddc01283ea4d6f/external/rules_nixpkgs_core/nixpkgs.bzl:795:41: in <toplevel>
ERROR: An error occurred during the fetch of repository 'nix_rust':
   Traceback (most recent call last):
	File "/private/var/tmp/_bazel_dwagnerhall/ce461197ed3d17b80eddc01283ea4d6f/external/rules_nixpkgs_core/nixpkgs.bzl", line 793, column 31, in _nixpkgs_flake_package_impl
		_nixpkgs_build_and_symlink(repository_ctx, [nix_path, "build"], expr_args, build_file_content)
	File "/private/var/tmp/_bazel_dwagnerhall/ce461197ed3d17b80eddc01283ea4d6f/external/rules_nixpkgs_core/nixpkgs.bzl", line 447, column 34, in _nixpkgs_build_and_symlink
		exec_result = execute_or_fail(
	File "/private/var/tmp/_bazel_dwagnerhall/ce461197ed3d17b80eddc01283ea4d6f/external/rules_nixpkgs_core/util.bzl", line 97, column 13, in execute_or_fail
		fail("""
Error in fail: 
  Cannot build Nix derivation for package '@nix_rust'.
    Command: "/nix/var/nix/profiles/default/bin/nix" "build" "path:/Volumes/Source/github.com/bazelbuild/rules_rust/examples/nix_cross_compiling/nix#bazel.rust" "--print-out-paths" "--out-link" "bazel-support/nix-out-link"
    Return code: 1
    Error output: 
      > error: flake 'path:/Volumes/Source/github.com/bazelbuild/rules_rust/examples/nix_cross_compiling/nix' does not provide attribute 'packages.aarch64-darwin.bazel.rust', 'legacyPackages.aarch64-darwin.bazel.rust' or 'bazel.rust'

ERROR: /Volumes/Source/github.com/bazelbuild/rules_rust/examples/nix_cross_compiling/WORKSPACE.bazel:17:17: fetching _nixpkgs_flake_package rule //external:nix_rust: Traceback (most recent call last):
	File "/private/var/tmp/_bazel_dwagnerhall/ce461197ed3d17b80eddc01283ea4d6f/external/rules_nixpkgs_core/nixpkgs.bzl", line 793, column 31, in _nixpkgs_flake_package_impl
		_nixpkgs_build_and_symlink(repository_ctx, [nix_path, "build"], expr_args, build_file_content)
	File "/private/var/tmp/_bazel_dwagnerhall/ce461197ed3d17b80eddc01283ea4d6f/external/rules_nixpkgs_core/nixpkgs.bzl", line 447, column 34, in _nixpkgs_build_and_symlink
		exec_result = execute_or_fail(
	File "/private/var/tmp/_bazel_dwagnerhall/ce461197ed3d17b80eddc01283ea4d6f/external/rules_nixpkgs_core/util.bzl", line 97, column 13, in execute_or_fail
		fail("""
Error in fail: 
  Cannot build Nix derivation for package '@nix_rust'.
    Command: "/nix/var/nix/profiles/default/bin/nix" "build" "path:/Volumes/Source/github.com/bazelbuild/rules_rust/examples/nix_cross_compiling/nix#bazel.rust" "--print-out-paths" "--out-link" "bazel-support/nix-out-link"
    Return code: 1
    Error output: 
      > error: flake 'path:/Volumes/Source/github.com/bazelbuild/rules_rust/examples/nix_cross_compiling/nix' does not provide attribute 'packages.aarch64-darwin.bazel.rust', 'legacyPackages.aarch64-darwin.bazel.rust' or 'bazel.rust'

ERROR: Error computing the main repository mapping: Encountered error while reading extension file 'defs.bzl': no such package '@@crate_index//': no such package '@@cargo_bazel_bootstrap//': no such package '@@nix_rust//': 
  Cannot build Nix derivation for package '@nix_rust'.
    Command: "/nix/var/nix/profiles/default/bin/nix" "build" "path:/Volumes/Source/github.com/bazelbuild/rules_rust/examples/nix_cross_compiling/nix#bazel.rust" "--print-out-paths" "--out-link" "bazel-support/nix-out-link"
    Return code: 1
    Error output: 
      > error: flake 'path:/Volumes/Source/github.com/bazelbuild/rules_rust/examples/nix_cross_compiling/nix' does not provide attribute 'packages.aarch64-darwin.bazel.rust', 'legacyPackages.aarch64-darwin.bazel.rust' or 'bazel.rust'


```